tests/worldmap/QuadTreeTest.cpp

```diff
@@ -0,0 +1,233 @@
+#include "../../src/worldmap/QuadTree.h"
+
+#include <iostream>
+
+#include <catch2/catch.hpp>
+
+namespace
+{
+double randNum(double low, double high)
+{
+	return low + (rand() / (RAND_MAX / (high - low))); // NOLINT(cert-msc50-cpp)
+}
+
+point_t randPoint(double size)
+{
+	return {randNum(-size / 2, size / 2), randNum(-size / 2, size / 2), 1};
+}
+} // namespace
+
+TEST_CASE("QuadTree - AddPoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
+	QuadTree tree(100);
+
+	REQUIRE(tree.empty());
+
+	for (int i = 0; i < 50; i++)
+	{
+		const point_t &point = randPoint(100);
+		REQUIRE(tree.getSize() == i);
+		REQUIRE(tree.add(point));
+		REQUIRE(tree.getSize() == i + 1);
+	}
+
+	REQUIRE_FALSE(tree.empty());
+}
+
+TEST_CASE("QuadTree - RemovePoints", "[QuadTree]")
+{
+	srand(time(nullptr)); // NOLINT(cert-msc51-cpp)
```

Comment:
Does this NOLINT comment have some effect on our testing suite?

Reply:
The main reason for putting it was my IDE was just yelling at me since clang-tidy doesn't like it when you use srand and rand instead of the newer random functions. I don't think it has any other side effects besides silencing warnings, right?